diff -urpN udk-toolchain-20140321.orig/mingw-gcc-build.py udk-toolchain-20140321/mingw-gcc-build.py
--- udk-toolchain-20140321.orig/mingw-gcc-build.py	2014-02-26 00:31:32.000000000 -0800
+++ udk-toolchain-20140321/mingw-gcc-build.py	2014-04-29 04:19:19.073667501 -0700
@@ -298,40 +298,20 @@ class SourceFiles:
                             print '[md5 match]',
                         else:
                             print '[md5 mismatch]',
+                            print '[failed]'
+                            print '  Tried to retrieve', url
+                            print '  to', local_file
+                            print 'Possible fixes:'
+                            print '* If you are behind a web-proxy, try setting the',
+                            print 'http_proxy environment variable'
+                            print '* You can try to download this file separately',
+                            print 'and rerun this script'
+                            raise Exception()
                         sys.stdout.flush()
                         completed = md5_pass
 
-                    if not completed:
-                        urllib.urlretrieve(url, local_file, progress)
-
-                    #
-                    # BUGBUG: Suggest proxy to user if download fails.
-                    #
-                    # export http_proxy=http://proxyservername.mycompany.com:911
-                    # export ftp_proxy=http://proxyservername.mycompany.com:911
-
-                    if not completed and os.path.exists(local_file):
-                        md5_pass = self.checkHash(fdata)
-                        if md5_pass:
-                            print '[md5 match]',
-                        else:
-                            print '[md5 mismatch]',
-                        sys.stdout.flush()
-                        completed = md5_pass
-
-                    if completed:
-                        print '[done]'
-                        break
-                    else:
-                        print '[failed]'
-                        print '  Tried to retrieve', url
-                        print '  to', local_file
-                        print 'Possible fixes:'
-                        print '* If you are behind a web-proxy, try setting the',
-                        print 'http_proxy environment variable'
-                        print '* You can try to download this file separately',
-                        print 'and rerun this script'
-                        raise Exception()
+                    print '[done]'
+                    break
                 
                 except KeyboardInterrupt:
                     print '[KeyboardInterrupt]'
